ruby/ext/google/protobuf_c/convert.c

```diff
@@ -164,7 +164,7 @@ upb_MessageValue Convert_RubyToUpb(VALUE value, const char* name,
       VALUE utf8 = rb_enc_from_encoding(rb_utf8_encoding());
       if (rb_obj_class(value) == rb_cSymbol) {
         value = rb_funcall(value, rb_intern("to_s"), 0);
-      } else if (rb_obj_class(value) != rb_cString) {
+      } else if (rb_obj_class(value) != rb_cString && !rb_obj_is_kind_of(value, rb_cString)) {
```

Comment:
Are both terms of the conditional check needed? `'foo'.kind_of?(String)` is `true`, so doesn't it cover both the case where `value` is a `String` and cases where it is is a subclass?

Reply:
You're right. 
✅ [2a6b513](https://github.com/protocolbuffers/protobuf/pull/13818/commits/2a6b513d81ff1d028987e98871ff0204e5bbea31)